src/worker.cc

```diff
@@ -1011,31 +1122,75 @@ std::string jsonify_error(const valhalla_exception_t& exception, Api& request) {
   err_stat->set_value(1);
   err_stat->set_type(count);
 
+  // pbf format output
+  if (request.options().format() == Options::pbf) {
+    return request.SerializeAsString();
+  }
+
+  // json
   return body.str();
 }
 
+void ParseApi(const std::string& request, Options::Action action, valhalla::Api& api) {
+  // if its a pbf request we only want to clear the stuff we are going to eventually set
+  if (api.has_options() && request.empty()) {
+    api.clear_trip();
+    api.clear_directions();
+    api.clear_status();
+    api.clear_info();
+  } // otherwise blank it all just in case
+  else {
+    api.Clear();
+  }
```

Comment:
this is a little tricky but the comment describes it. i just wanted to point it out